lib/typing/parse_timestamp.go

```diff
@@ -94,11 +96,24 @@ func ParseTimestampTZFromAny(val any) (time.Time, error) {
 		return convertedVal, nil
 	case string:
 		return parseTimestampTZ(convertedVal)
+	case float64:
+		value, err := primitives.Int64Converter{}.Convert(val)
```

Comment:
Our primitives Int64 converter will check for precision loss and if there are any, it'll error.